model/buildlogger.go

```diff
@@ -299,6 +301,42 @@ func (l *Log) Download(timeRange util.TimeRange) (LogIterator, error) {
 	return NewBatchedLogIterator(bucket, l.Artifact.Chunks, 100, timeRange), nil
 }
 
+// MergeLogs merges N logs, passed in as LogIterators, respecting the order of
+// each line's timestamp. Note that once all lines are merged, the string
+// channel is closed.
+func MergeLogs(ctx context.Context, iterators ...LogIterator) chan string {
+	h := &LogIteratorHeap{}
+	heap.Init(h)
+	lines := make(chan string, len(iterators))
+
+	for _, it := range iterators {
+		if it.Next(ctx) {
+			heap.Push(h, it)
+		}
+	}
+
+	go func() {
+		defer close(lines)
+		for h.Len() > 0 {
+			if ctx.Err() != nil {
+				return
+			}
+
+			i := heap.Pop(h)
+			it := i.(LogIterator)
+
+			line := fmt.Sprintf("%s %s", it.Item().Timestamp, it.Item().Data)
```

Comment:
I spoke to Max, and it looks like we can actually throw away the buildlogger generated timestamp after sorting (resmoke has its own timestamps in their logs). I kept here it for now since it may be useful  in other contexts (and easy enough to strip away or modify).